src/AvaTax.TaxModule.Web/Controller/AvaTaxController.cs

```diff
@@ -1,160 +1,163 @@
 using AvaTax.TaxModule.Core.Models;
 using AvaTax.TaxModule.Core.Services;
+using AvaTax.TaxModule.Data.Logging;
 using AvaTax.TaxModule.Web.BackgroundJobs;
 using AvaTax.TaxModule.Web.Models;
 using AvaTax.TaxModule.Web.Models.PushNotifications;
+using AvaTax.TaxModule.Web.Services;
 using Hangfire;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
 using VirtoCommerce.Platform.Core.PushNotifications;
 using VirtoCommerce.Platform.Core.Security;
-using VirtoCommerce.Platform.Core.Web.Security;
+using Microsoft.Extensions.Options;
 
 namespace AvaTax.TaxModule.Web.Controller
 {
     [ApiExplorerSettings(IgnoreApi = true)]
-    [RoutePrefix("api/tax/avatax")]
-    public class AvaTaxController : ApiController
+    [Route("api/tax/avatax")]
+    public class AvaTaxController : Microsoft.AspNetCore.Mvc.Controller
     {
         private readonly Func<IAvaTaxSettings, AvaTaxClient> _avaTaxClientFactory;
         private readonly AvalaraLogger _logger;
         private readonly IOrdersSynchronizationService _ordersSynchronizationService;
         private readonly IAddressValidationService _addressValidationService;
         private readonly IPushNotificationManager _pushNotificationManager;
         private readonly IUserNameResolver _userNameResolver;
-        
+        private readonly AvaTaxSecureOptions _options;
+
         [CLSCompliant(false)]
-        public AvaTaxController(ILog log, Func<IAvaTaxSettings, AvaTaxClient> avaTaxClientFactory, IOrdersSynchronizationService ordersSynchronizationService,
-            IAddressValidationService addressValidationService, IPushNotificationManager pushNotificationManager, IUserNameResolver userNameResolver)
+        public AvaTaxController(ILogger<AvaTaxController> log, Func<IAvaTaxSettings, AvaTaxClient> avaTaxClientFactory, IOrdersSynchronizationService ordersSynchronizationService,
+            IAddressValidationService addressValidationService, IPushNotificationManager pushNotificationManager, IUserNameResolver userNameResolver, IOptions<AvaTaxSecureOptions> options)
         {
             _logger = new AvalaraLogger(log);
             _avaTaxClientFactory = avaTaxClientFactory;
             _ordersSynchronizationService = ordersSynchronizationService;
             _addressValidationService = addressValidationService;
             _pushNotificationManager = pushNotificationManager;
             _userNameResolver = userNameResolver;
+            _options = options.Value;
         }
 
         [HttpPost]
-        [ResponseType(typeof(void))]
         [Route("ping")]
-        public IHttpActionResult TestConnection([FromBody]AvaTaxSettings taxSetting)
+        [ProducesResponseType(typeof(void), StatusCodes.Status204NoContent)]
+        public Task<ActionResult> TestConnection([FromBody]AvaTaxSettings taxSetting)
```

Comment:
Please add another all possible ResponseTypes, e.g. for BadRequest, Ok.
Ensure that it is set in all controller methods.

Reply:
done